### ISSUE TITLE
uninstaller: GCP teardown does not delete VPC?

### ISSUE THREAD

**Issue by sreis** [opstrace/opstrace#1318]
**Describe the bug**

While triaging #1317 I noticed that we've started to leak VPCs. In project `ci-shard-bbb` we have:

```
> gcloud compute networks list
NAME                    SUBNET_MODE  BGP_ROUTING_MODE  IPV4_RANGE  GATEWAY_IPV4
default                 AUTO         REGIONAL
prs-bk-5558-0be-ug      CUSTOM       REGIONAL
prs-bk-5563-0ad-ug      CUSTOM       REGIONAL
prs-bk-5579-906-ug      CUSTOM       REGIONAL
prs-bk-5596-46b-ug      CUSTOM       REGIONAL
prs-bk-5598-56b-ug      CUSTOM       REGIONAL
prs-bk-5607-204-ug      CUSTOM       REGIONAL
prs-bk-5609-572-g       CUSTOM       REGIONAL
schedul-bk-2885-b25-ug  CUSTOM       REGIONAL
schedul-bk-2888-11f-ug  CUSTOM       REGIONAL
schedul-bk-2892-11f-ug  CUSTOM       REGIONAL
schedul-bk-2896-fd1-ug  CUSTOM       REGIONAL
schedul-bk-2897-fd1-ug  CUSTOM       REGIONAL
schedul-bk-2899-b00-ug  CUSTOM       REGIONAL
schedul-bk-2901-b00-ug  CUSTOM       REGIONAL
schedul-bk-2905-ef3-ug  CUSTOM       REGIONAL
schedul-bk-2919-d1e-ug  CUSTOM       REGIONAL
schedul-bk-2926-d1e-ug  CUSTOM       REGIONAL
schedul-bk-2927-d1e-ug  CUSTOM       REGIONAL
schedul-bk-2932-d1e-ug  CUSTOM       REGIONAL
schedul-bk-2936-3b3-ug  CUSTOM       REGIONAL
schedul-bk-2937-3b3-ug  CUSTOM       REGIONAL
schedul-bk-2938-3b3-ug  CUSTOM       REGIONAL
upgrade-bk-1219-11f-g   CUSTOM       REGIONAL
upgrade-bk-1223-fd1-g   CUSTOM       REGIONAL
upgrade-bk-1231-ef3-g   CUSTOM       REGIONAL
upgrade-bk-1247-d1e-g   CUSTOM       REGIONAL
upgrade-bk-1248-d1e-g   CUSTOM       REGIONAL
upgrade-bk-1249-d1e-g   CUSTOM       REGIONAL
upgrade-bk-1250-d1e-g   CUSTOM       REGIONAL
```

Some of the builds ran the teardown successfully, (small sample):

https://buildkite.com/opstrace/scheduled-main-builds/builds/2937
https://buildkite.com/opstrace/scheduled-main-builds/builds/2938

https://buildkite.com/opstrace/upgrade-tests/builds/1249
https://buildkite.com/opstrace/upgrade-tests/builds/1250

**Comment by jgehrcke** [comment]
The stranded network that start with `schedul-*` all end with `-ug`, i.e. they come from the upgrade test job.

**Comment by sreis** [comment]
Another clue, the logs for the upgrade tests show that VPC deletion never occurs because [this message is not printed](https://github.com/opstrace/opstrace/blob/main/lib/gcp/src/network.ts#L243):

```
VPC deletion has started with status:
```

**Comment by sreis** [comment]
It looks like some of the failures are related to #976. I see the following error in a lot of the failed builds:

```
[2021-08-23T09:48:30Z] 2021-08-23T09:48:30.730Z [32minfo[39m: Ensure Address deletion
[2021-08-23T09:48:30Z] 2021-08-23T09:48:30.960Z [32minfo[39m: Global Address is RESERVED
[2021-08-23T09:48:33Z] 2021-08-23T09:48:33.947Z [32minfo[39m: Global Address is RESERVED
[2021-08-23T09:48:34Z] 2021-08-23T09:48:34.415Z [31merror[39m: error during cluster teardown (attempt 1):
[2021-08-23T09:48:34Z] GaxiosError: The resource 'projects/ci-shard-bbb/global/addresses/google-managed-services-upgrade-bk-1250-d1e-g' is not ready
[2021-08-23T09:48:34Z]     at Gaxios._request (/snapshot/build/node_modules/gaxios/src/gaxios.ts:140:15)
[2021-08-23T09:48:34Z]     at runMicrotasks (<anonymous>)
[2021-08-23T09:48:34Z]     at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2021-08-23T09:48:34Z]     at JWT.requestAsync (/snapshot/build/node_modules/google-auth-library/build/src/auth/oauth2client.js:343:18)
[2021-08-23T09:48:34Z]     at deleteAddress (/snapshot/build/lib/gcp/build/globalAddress.js) {
[2021-08-23T09:48:34Z]   response: [Object],
[2021-08-23T09:48:34Z]   config: [Object],
[2021-08-23T09:48:34Z]   code: 400,
[2021-08-23T09:48:34Z]   errors: [Array]
[2021-08-23T09:48:34Z] }
```

But the teardown script error is being swallowed so we missed it. I'll follow up with a PR to handle the teardown failure.

**Comment by sreis** [comment]
Cleaned up CI resources and merged #1320 with a workaround to handle Global Address deletion.
